packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.test.ts

```diff
@@ -0,0 +1,387 @@
+  const nestedMentionTextRepresentation = 'Hello #éeveryone!';
+  const nestedMentionTags: TagData[] = [
+    {
+      closingTagIndex: 83,
+      content: '<b>Hello</b> <msft-mention id="1" displayText="Everyone">everyone</msft-mention>',
+      openTagBody: '<p>',
+      openTagIndex: 0,
+      plainTextBeginIndex: 0,
+      plainTextEndIndex: 16,
+      subTags: [
+        {
+          closingTagIndex: 8,
+          content: 'Hello',
+          openTagBody: '<b>',
+          openTagIndex: 0,
+          plainTextBeginIndex: 0,
+          plainTextEndIndex: 5,
+          tagType: 'b'
+        },
+        {
+          closingTagIndex: 65,
+          content: 'everyone',
+          openTagBody: '<msft-mention id="1" displayText="Everyone">',
+          openTagIndex: 13,
+          plainTextBeginIndex: 6,
+          plainTextEndIndex: 16,
+          tagType: 'msft-mention'
+        }
+      ],
+      tagType: 'p'
+    },
+    {
+      closingTagIndex: 92,
+      content: '!',
+      openTagBody: '<em>',
+      openTagIndex: 87,
+      plainTextBeginIndex: 16,
+      plainTextEndIndex: 17,
+      tagType: 'em'
+    }
+  ];
+
+  const twoWordMention = 'Hello <msft-mention id="1" displayText="Patricia Adams">Patricia Adams</msft-mention>!';
+  const twoWordMentionTextRepresentation = 'Hello @Patricia Adams!';
+  const twoWordMentionTag: TagData = {
+    closingTagIndex: 70,
+    content: 'Patricia Adams',
+    openTagBody: '<msft-mention id="1" displayText="Patricia Adams">',
+    openTagIndex: 6,
+    plainTextBeginIndex: 6,
+    plainTextEndIndex: 21,
+    tagType: 'msft-mention'
+  };
+
+  test('Basic parsing works', () => {
+    const parsed = textToTagParser(basicMention, '@');
+
+    expect(parsed.plainText).toEqual(basicMentionTextRepresentation);
+    expect(parsed.tags).toEqual([basicMentionTag]);
+  });
+
+  test('Nested tags parsing and trigger of 2 characters succeeds', () => {
+    const parsed = textToTagParser(nestedMention, '#é');
+    expect(parsed.plainText).toEqual(nestedMentionTextRepresentation);
+    expect(parsed.tags).toEqual(nestedMentionTags);
+  });
+
+  test('Bad HTML does not break parsing', () => {
+    const badString = '<b>Hello, <i>world!</i>';
+    const plainText = 'Hello, world!';
+    let parsed: { tags: TagData[]; plainText: string } | undefined;
+    expect(() => {
+      parsed = textToTagParser(badString, '@');
+    }).not.toThrow();
+    expect(parsed?.plainText).toEqual(plainText);
+  });
+
+  test('Using < in the text does not break parsing', () => {
+    const comparisonString = 'x<y and z';
+    let parsed: { tags: TagData[]; plainText: string } | undefined;
+    expect(() => {
+      parsed = textToTagParser(comparisonString, '@');
+    }).not.toThrow();
+    expect(parsed?.plainText).toEqual(comparisonString);
+  });
+
+  test('Using > in the text does not break parsing', () => {
+    const comparisonString = 'x>y and z';
```

Comment:
Should we also add a test for `/>`?

Reply:
We should probably add that when we get the handling of the input of < and > working